### PR TITLE
CLN - Change string formatting in plotting

### PR DESCRIPTION
**Pull request by ohad83**
Changed the string formatting in pandas/plotting/*.py to f-strings.

- [x] tests passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

ref #29547

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -796,14 +793,12 @@ def __call__(self, *args, **kwargs):
...
-                    ("plot kind {} can only be used for data frames").format(kind)
-                )
+                raise ValueError(f"plot kind {kind} can only be used for data frames")
         elif kind in self._series_kinds:
             if isinstance(data, ABCDataFrame):
                 if y is None and kwargs.get("subplots") is False:
-                    msg = "{} requires either y column or 'subplots=True'"
-                    raise ValueError(msg.format(kind))
+                    msg = f"{kind} requires either y column or 'subplots=True'"
+                    raise ValueError(msg)
```
Comment (topper-123) [inline pandas/plotting/_core.py]: No need for an intermediate ``msg`` here, just put the message inside ValueError.

File: pandas/plotting/_matplotlib/boxplot.py [hunk 2]
```diff
@@ -75,8 +75,8 @@ def _validate_color_args(self):
                     if key not in valid_keys:
                         raise ValueError(
                             "color dict contains invalid "
-                            "key '{0}' "
-                            "The key must be either {1}".format(key, valid_keys)
+                            f"key '{key}' "
+                            f"The key must be either {valid_keys}"
```
Comment (topper-123) [inline pandas/plotting/_matplotlib/boxplot.py]: These two should just be on the same line.  Also add "." after ``'{key}'``.

File: pandas/plotting/_matplotlib/boxplot.py [hunk 3]
```diff
@@ -269,8 +269,8 @@ def _get_colors():
                     else:
                         raise ValueError(
                             "color dict contains invalid "
-                            "key '{0}' "
-                            "The key must be either {1}".format(key, valid_keys)
+                            f"key '{key}' "
+                            f"The key must be either {valid_keys}"
```
Comment (topper-123) [inline pandas/plotting/_matplotlib/boxplot.py]: Same here

File: pandas/plotting/_matplotlib/timeseries.py [hunk 4]
```diff
@@ -307,7 +307,7 @@ def _maybe_convert_index(ax, data):
 
 
 def _format_coord(freq, t, y):
-    return "t = {0}  y = {1:8f}".format(Period(ordinal=int(t), freq=freq), y)
+    return f"t = {Period(ordinal=int(t), freq=freq)}  y = {y:8f}"
```
Comment (topper-123) [inline pandas/plotting/_matplotlib/timeseries.py]: IMO there is too much packed in this f-string. Could you put the ``Period(..)`` part in a varable and call that in the f-string?

**Review by topper-123 — CHANGES_REQUESTED** [review]
Looks good, just a few comments.

File: pandas/plotting/_matplotlib/boxplot.py (hunk 2 above)
Comment (ohad83) [reply pandas/plotting/_matplotlib/boxplot.py]: Agreed, though I moved the key line to the previous line, makes it more readable.

**Review by topper-123 — APPROVED** [review]

**Comment by topper-123** [comment]
Thanks, @ohad83!

**Pull request merged**